API search request fails due to missing property in elasticsearch client

### <!-- Component -->

<!-- Data Layer- -->

### <!-- Affected version -->

<!-- -8.9 -->

### <!-- Severity -->

<!-- Unknown- -->

### <!-- Likelihood -->

<!-- Unknown_ -->

### Description

A Camunda API search request fails to query the Elasticsearch secondary storage due to a missing Elasticsearch client configuration property that is required.

### Steps to reproduce

Unknown steps. Observed in:
- Environment: Camunda SaaS
- Version: Camunda `8.9.0-alpha4`
- [Downloaded GCP logs (internal)](https://drive.google.com/file/d/1A33P2wdMrdXJ8lzlFjkoGg79THLbNl-Q/view?usp=sharing)
- [GCP logs viewer link (internal)
](https://console.cloud.google.com/logs/query;query=logName:%22stdout%22%0Aresource.type%3D%22k8s_container%22%0Aresource.labels.container_name%3D%22unified-gateway%22%0Aresource.labels.project_id%3D%22camunda-saas-dev%22%0Aresource.labels.cluster_name%3D%22worker-2%22%0Aresource.labels.location%3D%22europe-north1%22%0Aresource.labels.pod_name%3D%22unified-gateway-66644457bb-2sg8s%22%0Aresource.labels.namespace_name%3D%22097b5f7a-112a-4592-ae1c-e1855549c383-zeebe%22;pinnedLogId=2026-02-13T07:59:19.689796452Z%2Fj5vu70driqktffs9;cursorTimestamp=2026-02-13T07:59:19.689796452Z;startTime=2026-02-13T07:29:49.689Z;endTime=2026-02-13T08:29:49.689Z?project=camunda-saas-dev)

### Current behavior

The search request fails.

### Expected behavior

The search request should succeed.

### Workaround

1. do this
2. do that
3. now this happens and circumvents the bug


### Environment

SaaS

### Version

- Component Version: Unified Gateway `8.9.0-alpha4`


### Rootcause

Seems like the `ShardFailure.shard` Elasticsearch client required configuration property is missing.

This can be seen in the stacktrace:
```
io.camunda.search.exception.CamundaSearchException: Failed to execute index request
	at io.camunda.search.es.clients.ElasticsearchSearchClient.index(ElasticsearchSearchClient.java:184)
	at io.camunda.search.clients.PersistentWebSessionSearchImpl.upsertPersistentWebSession(PersistentWebSessionSearchImpl.java:46)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at io.camunda.authentication.session.WebSessionRepository.saveWebSessionIfChanged(WebSessionRepository.java:104)
	at io.camunda.authentication.session.WebSessionRepository.save(WebSessionRepository.java:52)
	at io.camunda.authentication.session.WebSessionRepository.save(WebSessionRepository.java:19)
	at org.springframework.session.web.http.SessionRepositoryFilter$SessionRepositoryRequestWrapper.commitSession(SessionRepositoryFilter.java:229)
	at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:145)
	at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:82)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:362)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:278)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:138)
	at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:110)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:138)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:138)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:165)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:88)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:113)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:83)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:72)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:733)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:903)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1774)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:973)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:491)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: co.elastic.clients.transport.TransportException: node: http://elasticsearch-es-http:9200/, status: 201, [es/index] Failed to decode response
	at co.elastic.clients.transport.ElasticsearchTransportBase.decodeTransportResponse(ElasticsearchTransportBase.java:409)
	at co.elastic.clients.transport.ElasticsearchTransportBase.getApiResponse(ElasticsearchTransportBase.java:367)
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:148)
	at co.elastic.clients.elasticsearch.ElasticsearchClient.index(ElasticsearchClient.java:1312)
	at io.camunda.search.es.clients.ElasticsearchSearchClient.index(ElasticsearchSearchClient.java:178)
	... 37 more
Caused by: co.elastic.clients.json.JsonpMappingException: Error deserializing co.elastic.clients.elasticsearch._types.ShardStatistics: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'ShardFailure.shard' (JSON path: _shards.failures[0]) (line no=1, column no=644, offset=-1)
	at co.elastic.clients.json.JsonpMappingException.from0(JsonpMappingException.java:134)
	at co.elastic.clients.json.JsonpMappingException.from(JsonpMappingException.java:125)
	at co.elastic.clients.json.JsonpDeserializerBase$ArrayDeserializer.deserialize(JsonpDeserializerBase.java:322)
	at co.elastic.clients.json.JsonpDeserializerBase$ArrayDeserializer.deserialize(JsonpDeserializerBase.java:280)
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77)
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:78)
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:214)
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:166)
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:157)
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77)
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:79)
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43)
	at co.elastic.clients.json.ObjectDeserializer$FieldObjectDeserializer.deserialize(ObjectDeserializer.java:78)
	at co.elastic.clients.json.ObjectDeserializer.deserializeInner(ObjectDeserializer.java:214)
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:166)
	at co.elastic.clients.json.ObjectDeserializer.deserialize(ObjectDeserializer.java:157)
	at co.elastic.clients.json.JsonpDeserializer.deserialize(JsonpDeserializer.java:77)
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:79)
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:43)
	at co.elastic.clients.transport.ElasticsearchTransportBase.decodeTransportResponse(ElasticsearchTransportBase.java:404)
	... 41 more
Caused by: co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'ShardFailure.shard'
	at co.elastic.clients.util.ApiTypeHelper.requireNonNull(ApiTypeHelper.java:76)
	at co.elastic.clients.elasticsearch._types.ShardFailure.<init>(ShardFailure.java:83)
	at co.elastic.clients.elasticsearch._types.ShardFailure.<init>(ShardFailure.java:62)
	at co.elastic.clients.elasticsearch._types.ShardFailure$Builder.build(ShardFailure.java:251)
	at co.elastic.clients.elasticsearch._types.ShardFailure$Builder.build(ShardFailure.java:176)
	at co.elastic.clients.json.ObjectBuilderDeserializer.deserialize(ObjectBuilderDeserializer.java:86)
	at co.elastic.clients.json.DelegatingDeserializer$SameType.deserialize(DelegatingDeserializer.java:48)
	at co.elastic.clients.json.JsonpDeserializerBase$ArrayDeserializer.deserialize(JsonpDeserializerBase.java:318)
	... 58 more

```

### Solution ideas

_No response_

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### Links

_No response_

### Automated Test Impact

_No response_

--- a/search/search-client-elasticsearch/src/test/java/io/camunda/search/es/clients/ElasticsearchDataStoreClientTest.java
+++ b/search/search-client-elasticsearch/src/test/java/io/camunda/search/es/clients/ElasticsearchDataStoreClientTest.java
@@ -15,6 +15,7 @@ import static org.mockito.Mockito.when;
 
 import co.elastic.clients.elasticsearch.ElasticsearchClient;
 import co.elastic.clients.elasticsearch._types.Result;
+import co.elastic.clients.elasticsearch._types.ShardFailure;
 import co.elastic.clients.elasticsearch.core.GetRequest;
 import co.elastic.clients.elasticsearch.core.GetResponse;
 import co.elastic.clients.elasticsearch.core.IndexRequest;
@@ -31,6 +32,7 @@ import io.camunda.search.clients.core.SearchWriteResponse;
 import io.camunda.search.es.transformers.ElasticsearchTransformers;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.mockito.ArgumentCaptor;
@@ -170,6 +172,27 @@ public class ElasticsearchDataStoreClientTest {
   }
 
   @Test
+  public void shouldTransformIndexResponseWithShardFailures() throws IOException {
+    // given - an index response with shard failures where the "shard" field is null
+    final var indexRequestCaptor = ArgumentCaptor.forClass(IndexRequest.class);
+    final var indexResponse = createIndexResponseWithShardFailures();
+    when(client.index(indexRequestCaptor.capture())).thenReturn(indexResponse);
+
+    final var doc = new TestDocument("test");
+    final var searchIndexRequest =
+        SearchIndexRequest.of(b -> b.id("foo").index("bar").document(doc));
+
+    // when
+    final var response = searchClient.index(searchIndexRequest);
+
+    // then -- should succeed despite no shard field in shard failures response
+    assertThat(response).isNotNull();
+    assertThat(response.id()).isEqualTo("foo");
+    assertThat(response.index()).isEqualTo("bar");
+    assertThat(response.result()).isEqualTo(SearchWriteResponse.Result.CREATED);
+  }
+
+  @Test
   public void shouldHaveMoreTotalItemsFalse() throws IOException {
     // given
     final var searchRequestCaptor = ArgumentCaptor.forClass(SearchRequest.class);
@@ -208,7 +231,7 @@ public class ElasticsearchDataStoreClientTest {
   }
 
   private SearchResponse<TestDocument> createDefaultSearchResponse(
-      long totalHits, TotalHitsRelation totalHitsRelation) {
+      final long totalHits, final TotalHitsRelation totalHitsRelation) {
     return SearchResponse.of(
         (f) ->
             f.took(122)
@@ -236,6 +259,31 @@ public class ElasticsearchDataStoreClientTest {
                 .seqNo(1L)
                 .version(1L)
                 .shards(s -> s.total(1).successful(1).failed(0)));
+  }
+
+  private IndexResponse createIndexResponseWithShardFailures() {
+    return IndexResponse.of(
+        b ->
+            b.id("foo")
+                .index("bar")
+                .result(Result.Created)
+                .primaryTerm(1L)
+                .seqNo(1L)
+                .version(1L)
+                .shards(
+                    s ->
+                        s.total(2)
+                            .successful(1)
+                            .failed(1)
+                            .failures(
+                                List.of(
+                                    ShardFailure.of(
+                                        sf ->
+                                            sf.reason(
+                                                    r ->
+                                                        r.type("i_o_exception")
+                                                            .reason("No space left on device"))
+                                                .status("INTERNAL_SERVER_ERROR"))))));
   }
 
   record TestDocument(String id) {}

